src/irr/asset/CGLSLBRDFBuiltinIncludeLoader.h

```diff
@@ -150,33 +226,28 @@ irr_glsl_ViewSurfaceInteraction  irr_glsl_calcRaySurfaceInteraction(in irr_glsl_
 */
 
 // will normalize all the vectors
-irr_glsl_BSDFIsotropicParams irr_glsl_calcBSDFIsotropicParams(in irr_glsl_ViewSurfaceInteraction interaction, in vec3 L)
+irr_glsl_BSDFIsotropicParams irr_glsl_calcBSDFIsotropicParams(in irr_glsl_IsotropicViewSurfaceInteraction interaction, in vec3 L)
 {
-   float invlenV2 = inversesqrt(dot(interaction.V.dir,interaction.V.dir));
-   float invlenN2 = inversesqrt(dot(interaction.N,interaction.N));
    float invlenL2 = inversesqrt(dot(L,L));
 
    irr_glsl_BSDFIsotropicParams params;
 
    // totally useless vectors, will probably get optimized away by compiler if they don't get used
    // but useful as temporaries
-   params.interaction.V.dir = interaction.V.dir*invlenV2;
-   params.interaction.N = interaction.N*invlenN2;
+   params.interaction = interaction;
    params.L = L*invlenL2;
    params.invlenL2 = invlenL2;
 
    // this stuff only works with normalized L,N,V
    params.NdotL = dot(params.interaction.N,params.L);
    params.NdotL_squared = params.NdotL*params.NdotL;
-   params.NdotV = dot(params.interaction.N,params.interaction.V.dir);
-   params.NdotV_squared = params.NdotV*params.NdotV;
 
    params.VdotL = dot(params.interaction.V.dir,params.L);
    float LplusV_rcpLen = inversesqrt(2.0 + 2.0*params.VdotL);
    params.LplusV_rcpLen = LplusV_rcpLen;
 
    // this stuff works unnormalized L,N,V
-   params.NdotH = (params.NdotL+params.NdotV)*LplusV_rcpLen;
+   params.NdotH = (params.NdotL+params.interaction.NdotV)*LplusV_rcpLen;
    params.VdotH = LplusV_rcpLen + LplusV_rcpLen*params.VdotL;
 
    return params;
```

Comment:
For `irr_glsl_BSDFAnisotropicParams irr_glsl_calcBSDFAnisotropicParams`, just require an anisotropic surface interaction and `L` instead of what we have already

Also add overloads for both `irr_glsl_calcBSDFIsotropicParams` and `irr_glsl_calcBSDFAnisotropicParams` that take `BSDFSample` instead of `L`

Reply:
let me know if tackled in new implementation.

Reply:
?